ethereum-types/src/hash.rs

```diff
@@ -68,14 +68,14 @@ impl_serde!(H264, 33);
 impl_serde!(H512, 64);
 impl_serde!(H520, 65);
 
-construct_hash!(H32, 4);
-construct_hash!(H64, 8);
-construct_hash!(H128, 16);
-construct_hash!(H160, 20);
-construct_hash!(H256, 32);
-construct_hash!(H264, 33);
-construct_hash!(H512, 64);
-construct_hash!(H520, 65);
+construct_fixed_hash!{ pub struct H32(4); }
```

Comment:
I know this is a lot to ask for, but I for one would really appreciate a rough idea of what we're using all of these hash types for. Are they all in use? Can we deprecate some? Not blocking this PR of course, just something worth pondering.

Reply:
With the new `construct_fixed_hash` macro we are able to mark them some as deprecated. :)

Reply:
The problem is that deprecating is not a viable option unless we want to have our terminals full of warnings. Even deprecating only the `H32` was hilarious.